ccf_transaction_processor/transaction_processor/pdo_tp.cpp

```diff
@@ -51,6 +54,29 @@ namespace ccfapp
             return ccf::make_success(true);
         };
 
+        //======================================================================================================
+        // register enclave expected measurements (member method)
+        auto register_enclave_expected_measurements = [this](auto& ctx, const nlohmann::json& params) {
+
+            const auto in = params.get<RegisterEnclaveExpectedMeasurements::In>();
+
+            // get the current measurements view
+            auto measurements_view = ctx.tx.rw(enclave_expected_measurements);
+
+            // collect the data to be stored
+            EnclaveExpectedMeasurements expected_measurements;
+
+            expected_measurements.mrenclave = in.mrenclave;
+            expected_measurements.basename = in.basename;
+            expected_measurements.ias_public_key = in.ias_public_key;
+
+            //store the data
+            measurements_view->put(PDO_ENCLAVE_EXPECTED, expected_measurements);
+
+            return ccf::make_success(true);
+        };
+
+
         //======================================================================================================
```

Comment:
who gets to submit this transaction? and how do you make sure it only happens once? and what happens to the actual registration transactions before this is set?

Reply:
thank you for the question. "register_enclave_expected_measurements" is regsitered as a CCF member rpc via
       make_endpoint(
            REGISTER_ENCLAVE_EXPECTED_MEASUREMENTS,
            HTTP_POST,
            json_adapter(register_enclave_expected_measurements),
            member_cert_sign_required).install();

Since, this is a consortium rpc, the consortium gets to update the expected MREnclave whenever they deem it appropriate to do so. (The consortium will vote among themselves). of course, in our "research" setting, we only use a single member consortium.